lib/vnet/daemon/client_darwin.m

```diff
@@ -0,0 +1,99 @@
+// Teleport
+// Copyright (C) 2024 Gravitational, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#import <Foundation/Foundation.h>
+#import <ServiceManagement/ServiceManagement.h>
+
+#include "client_darwin.h"
+
+void BundlePath(struct BundlePathResult *result) {
+    // From the docs:
+    // > This method may return a valid bundle object even for unbundled apps.
+    // > It may also return nil if the bundle object could not be created,
+    // > so always check the return value.
+    NSBundle *main = [NSBundle mainBundle];
+    if (!main) {
+        result->bundlePath = strdup("");
+        return;
+    }
+    
+    result->bundlePath = VNECopyNSString([main bundlePath]);
+}
+
+NSString * DaemonLabel(void) {
+    NSBundle *main = [NSBundle mainBundle];
+    if (!main) {
+        return @"";
+    }
+    
+    NSString *bundleIdentifier = [main bundleIdentifier];
+    if ([bundleIdentifier length] == 0) {
+        return bundleIdentifier;
+    }
+    
+    return [NSString stringWithFormat:@"%@.vnetd", bundleIdentifier];
```

Comment:
I might need to adjust assumptions made about bundle identifiers here. This code works locally, but I just made a tag build and it doesn't work.

I assumed that in development, our bundle identifier is just `com.goteleport.tshdev` (as listed in Info.plist). However, it's actually `K497G57PDJ.com.goteleport.tshdev`.

I haven't checked yet if this is the root cause of the problem, since Connect simply hangs either when calling `C.DaemonStatus` or `C.RegisterDaemon`.

Reply:
Ok, so my assumptions were correct and it was an issue with cache. For some reason, macOS really doesn't like when I switch from a local signed build to a tag build. Weird, because both are signed with the same key.

`sfltool dumpbtm` returns this, hence why I think my assumptions about identifiers are correct. The bundle identifier _is_ `com.goteleport.tshdev`, so I think it's reasonable for the daemon plist to be identified as `com.goteleport.tshdev.vnetd`.

```
 #5:
                 UUID: D07CBD74-9A97-479C-986F-0C7644E0F5C9
                 Name: tsh
       Developer Name: Ada Lin
      Team Identifier: K497G57PDJ
                 Type: app (0x2)
          Disposition: [disabled, allowed, visible, notified] (10)
           Identifier: identifier "K497G57PDJ.com.goteleport.tshdev" and anchor apple generic and certificate 1[field.1.2.840.113635.100.6.2.6] /* exists */ and certificate leaf[field.1.2.840.113635.100.6.1.13] /* exists */ and certificate leaf[subject.OU] = K497G57PDJ
                  URL: file:///Applications/tsh.app/
           Generation: 0
    Bundle Identifier: com.goteleport.tshdev
  Embedded Item Identifiers:
    #1: com.goteleport.tshdev.vnetd

 #6:
                 UUID: 77943637-3F1F-429F-B118-E70D207D2AAA
                 Name: tsh
       Developer Name: (null)
                 Type: daemon (0x10)
          Disposition: [enabled, allowed, visible, notified] (11)
           Identifier: com.goteleport.tshdev.vnetd
                  URL: Contents/Library/LaunchDaemons/com.goteleport.tshdev.vnetd.plist
      Executable Path: Contents/MacOS/tsh
           Generation: 1
    Parent Identifier: identifier "K497G57PDJ.com.goteleport.tshdev" and anchor apple generic and certificate 1[field.1.2.840.113635.100.6.2.6] /* exists */ and certificate leaf[field.1.2.840.113635.100.6.1.13] /* exists */ and certificate leaf[subject.OU] = K497G57PDJ
```